### PR TITLE
Hide/show alerts depending on the data availability CASE-91

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**

### CODE REVIEW

File: app/javascript/Clients/clientIdPage/SafetyAlertInformation.js [hunk 1]
```diff
@@ -174,6 +191,12 @@ export default class SafetyAlertInformation extends React.Component {
     );
   }
 }
+function getCardHeaderText({ XHRStatus, records }, text) {
+  return XHRStatus === 'ready' && records && records.length
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/SafetyAlertInformation.js]: The results of this aren't tested.

File: app/javascript/Clients/clientIdPage/SafetyAlertInformation.js [hunk 2]
```diff
@@ -38,14 +39,16 @@ export default class SafetyAlertInformation extends React.Component {
...
+      .then(safetyAlerts => {
+        this.setState({
+          safetyAlerts: {
+            XHRStatus: 'ready',
+            records: safetyAlerts,
+          },
+        });
       })
-    );
+      .catch(() => this.setState({ safetyAlerts: { XHRStatus: 'error' } }));
```
[the remark this answers is not in the document]
Comment (Sushmareddy-OSI) [reply app/javascript/Clients/clientIdPage/SafetyAlertInformation.js]: done!

File: app/javascript/Clients/clientIdPage/SafetyAlertInformation.test.js [hunk 3]
```diff
@@ -36,6 +36,15 @@ describe('#setClients', () => {
         expect(getSafetyAlertSpy).toHaveBeenCalledWith();
         expect(getSafetyAlertSpy).toHaveBeenCalledTimes(2);
       });
+
+      it('tracks case api requests', () => {
+        ClientService.fetchSafetyAlerts.mockReturnValue(
```
[the remark this answers is not in the document]
Comment (Sushmareddy-OSI) [reply app/javascript/Clients/clientIdPage/SafetyAlertInformation.test.js]: This is the test for above catch(error). refactored!

File: app/javascript/Clients/clientIdPage/SafetyAlertInformation.js (hunk 1 above)
Comment (Sushmareddy-OSI) [reply app/javascript/Clients/clientIdPage/SafetyAlertInformation.js]: As discussed with @edgibbs , this function is tested in render method.

**Pull request merged**